scripts/vim-patch.sh

```diff
@@ -394,6 +394,22 @@ list_missing_vimpatches() {
   local token vim_commit vim_tag patch_number
   declare -A tokens
   declare -A vim_commit_tags
+  declare -a git_log_args
+
+  # Massage arguments for git-log.
+  declare -A git_log_replacements=(
+    [^\(.*/\)?src/nvim/\(.*\)]="\${BASH_REMATCH[1]}src/\${BASH_REMATCH[2]}"
+    [^\(.*/\)?\.vim-src/\(.*\)]="\${BASH_REMATCH[2]}"
+  )
+  for i in "$@"; do
+    for j in "${!git_log_replacements[@]}"; do
+      if [[ "$i" =~ $j ]]; then
+        eval "git_log_args+=(${git_log_replacements[$j]})"
+        continue 2
+      fi
+    done
+    git_log_args+=("$i")
+  done
```

Comment:
can this new code go in a separate function?

Reply:
Yes, in theory.  But given that it changes an array it would need to set it in the function then, with the `local` before the call to it (a pattern you did not like the last time).
Not sure though that it is worth it in general though - I think a code block with a comment on top is good enough.

Reply:
ok, I was thinking this is rather "noise" for this function. But if it's a pain then forget it. LGTM